dash/dash.py

```diff
@@ -352,16 +352,14 @@ def serve_routes(self):
             mimetype='application/json'
         )
 
+    # pylint: disable=too-many-branches
     def _collect_and_register_resources(self, resources):
         # now needs the app context.
         # template in the necessary component suite JS bundles
         # add the version number of the package as a query parameter
         # for cache busting
         def _relative_url_path(relative_package_path='', namespace=''):
 
-            # track the registered packages
-            self.registered_paths[namespace].add(relative_package_path)
-
             module_path = os.path.join(
```

Comment:
As suggested, moved the side-effect outside of the path calculator and calling it directly below